ls1intum/Artemis
# `General`: Allow tutors to view course exercise scores

### Checklist
#### General



#### Client

### Description

- Reverting client-sided changes to the `Scores` view, which was made unavailable for tutors and editors within [General: Enable score export with points per grading criterion into a CSV file #4135](https://github.com/ls1intum/Artemis/pull/4135).
Tutors _(and Editors)_ should be able to access the scores view for exercises, as this is useful for answering questions.
- The result details should always be displayed with the test names to tutors, which applies to the result view within the submission view.
- Minor changes:
  - Aligned buttons in `programming-exercise-detail-view`
  - Improved consistency for buttons that are displayed to tutors and editors
    - `Participation` button was missing in `programming-exercise-detail` for tutors
    - `Submissions` button was missing in `programming-exercise` for editors

### Steps for Testing

Prerequisites:
- 1 Tutor / Editor
- 1 Instructor

1. Tutor/Editor: Check the affected views, make sure that the `Scores` button is available.
2. Instructor: Check that you can view the `Submissions` for programming exercises _(and exam programming exercises - only applies to Instructors)_ and the result details contain the name of the test

Affected views:
_(https://github.com/ls1intum/Artemis/pull/4135/files and search for occurrences of `isAtLeast`)_
- `course-management-exercise-row`
- `file-upload-exercise.component`
- `modeling-exercise.component`
- `programming-exercise-detail`
- `quiz-exercise`
- `non-programming-exercise-detail-common-actions`
- `text-exercise-row-buttons`
- `course-exercise-details`

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots



<details>
<summary>Scores button</summary>

E.g. the `programming-exercise-detail` view without `Scores` button (editor view):
![image](https://user-images.githubusercontent.com/63976129/144059953-58ce164d-e4d8-48a4-9d0e-dfb34efd41b0.png)

Fixed view with `Scores button`:
_I also removed a margin from the div, so the buttons are aligned properly_
![image](https://user-images.githubusercontent.com/63976129/144060530-641c4d18-2b90-45b2-b1b3-8f193172df1b.png)
</details>

<details>
<summary>Submissions</summary>
Navigation to the submissions result details view (showing fixed version):

https://user-images.githubusercontent.com/63976129/144162177-d69002a3-7484-408d-a0ac-70fd57569c44.mp4

Submission result details **without** test name (broken version):
![image](https://user-images.githubusercontent.com/63976129/144162448-5db38980-db35-4691-b3e2-d0e941c9311d.png)

Submission result details **with** test name (fixed version):
![image](https://user-images.githubusercontent.com/63976129/144162243-5a3f6510-b2a5-4ed6-af95-af788394dc12.png)


</details>
